lib/sig0.js

```diff
@@ -97,6 +97,18 @@ sig0.createKey = function createKey(algorithm, publicKey) {
   rd.flags = 0;
   rd.protocol = 0;
   rd.algorithm = algorithm;
+
+  // https://tools.ietf.org/html/rfc4034#appendix-A.1.1
+  // PRIVATEDNS publicKey should be prefixed with algo name and null byte 
+  if (algorithm === sig0.algs.PRIVATEDNS) {
+    // const name = "secp256k1sha256";
+    const name = "";
+    const key = bio.write(name.length + publicKey.length + 1);
```

Comment:
Maybe `name.length + 1 + ...` to imply that the extra 1 byte comes from the null termination?

Reply:
Unfortunately there is no entry in `algsByVal` for this

https://github.com/chjj/bns/blob/master/lib/constants.js#L544

Reply:
Yeah I think this should be a const, though maybe not in `algsByVal`.

Reply:
I think best would be a object mapping a enum to names, and passing in the enum when creating the sig0 key. (similar to `PRIVATEDNS`).

Reply:
Keeping it simple for now. Might revisit later.